neuralmonkey/learning_utils.py

```diff
@@ -421,6 +421,15 @@ def run_on_dataset(tf_manager: TensorFlowManager,
                 if isinstance(data, np.ndarray):
                     np.save(path, data)
                     log("Result saved as numpy array to '{}'".format(path))
+                elif (isinstance(data, list)
```

Comment:
Tohle IHMO nestačí, chtělo by to ještě kontrolovat, že je do dict nampájů.

Reply:
to bych musel projíždět celej ten dict.. Nejradši bych na to použil `typeguard.check_type`, ale byl by to ugly hack, asi jako tohle. Koukni do komentáře v komitu, co jsem k tomu udělal, tam je example

Reply:
Mozna by stalo za to, checknout https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/util/nest.py
Ale nevim, jestli se to hodi zrovna na tento problem (nezkoumal jsem do detailu, co vsechno to podporuje)

Reply:
To je užitečný na procházení těch nested struktur, který se používaj třeba pro feed dict nebo pro fetches.. Tady chci jen čekovat typ...